File: ria/java/gof/src/main/java/com/gof/pattern/abstractFactory/after/WhiteShipPartsFactory.java
package com.gof.pattern.abstractFactory.after;

import com.gof.pattern.abstractFactory.before.WhiteAnchor;
import com.gof.pattern.abstractFactory.before.WhiteWheel;

public class WhiteShipPartsFactory implements ShipPartsFactory {
    @Override
    public Anchor createAnchor() {
        return new WhiteAnchor();
    }

    @Override
    public Wheel createWheel() {
        return new WhiteWheel();
    }
}

File: ria/gof/src/main/kotlin/com/gof/pattern/abstractFactory/after/WhiteShipPartsFactory.kt
package com.gof.pattern.abstractFactory.after

import com.gof.pattern.abstractFactory.before.WhiteAnchor
import com.gof.pattern.abstractFactory.before.WhiteWheel

class WhiteShipPartsFactory : ShipPartsFactory {

    override fun createAnchor(): Anchor {
        return WhiteAnchor()
    }

    override fun createWheel(): Wheel {
        return WhiteWheel()
    }
}
